Slim/Plugin/Podcast/ProtocolHandler.pm

```diff
@@ -0,0 +1,85 @@
+package Slim::Plugin::Podcast::ProtocolHandler;
+
+# Logitech Media Server Copyright 2005-2020 Logitech.
+
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License, 
+# version 2.
+
+use base qw(Slim::Player::Protocols::HTTPS);
+
+use strict;
+
+use Slim::Utils::Log;
+use Slim::Utils::Prefs;
+use Slim::Utils::Cache;
+
+my $log   = logger('plugin.podcast');
+my $prefs = preferences('plugin.podcast');
+my $cache = Slim::Utils::Cache->new;
+
+Slim::Player::ProtocolHandlers->registerHandler('podcast', __PACKAGE__);
+
+# remove podcast:// protocol to scan real url
+sub scanUrl {
+	my ( $class, $url, $args ) = @_;
+	my $track = $args->{song}->track;
+	my ($scanUrl, $from) = Slim::Plugin::Podcast::Plugin::unwrap($url);	
+	
+	main::INFOLOG && $log->info("Scanning podcast $url for title ", $track->title);
+
+	# as for redirect, need to port to new url the cover/icon set in XMLBrowser
+	if ( my $icon = $cache->get("remote_image_" . $url) ) {
+		$cache->set("remote_image_" . $scanUrl, $icon, '30 days');
+	}
+	
+	# current $track will be wiped out as $url changes so pass title
+	$args->{title} = $track->title;
+	$args->{song}->seekdata( { timeOffset => $from } ) if $from;
+	$class->SUPER::scanUrl($scanUrl, $args);
+}
+
+# we want to always be the protocol handler for the $song (not track)
+sub getSongHandler {
+	return __PACKAGE__;
+}
+
+sub new {
+	my ($class, $args) = @_;
+	my $song = $args->{song};	
+	my $startTime = $song->seekdata->{timeOffset} if $song->seekdata;
+	
+	main::INFOLOG && $log->info( "Streaming podcast $args->{url} from $startTime" );
+	
+	# erase last position from cache
+	$cache->remove('podcast-' . $1) if Slim::Plugin::Podcast::Plugin::unwrap($song->originUrl);

```

Comment:
Is `$1` set here from `unwrap()`? I'd have something learned again today! But I don't like it anyway, as it makes assumptions about the implementation of `unwrap()`. Please receive the result in a variable and use that instead.

Reply:
yes it is - but here I would disagree. Even if I add a variable, it's always an assumption on what unwrap returns, no matter what.
```
my ($url) = Slim::Plugin::Podcast::Plugin::unwrap($song->originUrl);
$cache->remove('podcast-' . $url) if $url
```
looks heavier to me and if unwrap changes, it fails anyway

Reply:
Maybe heavier, yes, but safer. `$1` isn't the result of `unwrap()` but of the regex being used inside that function, isn't it? Thus if you changed the implementation of `unwrap()` to not use a regex, or to structure it differently, `$1` would change or no longer be available.

Reply:
Yes you're right - I was thinking of the cases where I did a 
```
my $var = $1 if <some regex>
```
but that's different here